recipes/cmaes/all/patches/CompilerOptionsEigen.patch

```diff
@@ -0,0 +1,142 @@
++      "C4251"
++      "C4267" # conversion from 'size_t' to 'int', possible loss of data
++      "C4275"
++      "C4355"
++      "C4371" # layout of class may have changed from a previous version of the
++              # compiler due to better packing of member '...'
++      "C4388" # signed/unsigned mismatch
++      "C4296" # '>=' : expression is always true
++      "C4350" # behavior change: 'std::_Wrap_alloc...'
++      "C4365" # '=' : conversion from 'size_t' to 'int',
++              # signed/unsigned mismatch
++      "C4389" # '!=' : signed/unsigned mismatch
++      "C4464" # relative include path contains '..'
++      "C4510" # 'argument' : default constructor could not be generated
++      "C4571"
++      "C4512" # 'argument' : assignment operator could not be generated
++      "C4514" # 'function': unreferenced inline function has been removed
++      "C4548" # expression before comma has no effect; expected expression with
++              # side-effect" caused by FD_* macros.
++      "C4610" # struct 'argument' can never be instantiated - user defined
++              # constructor required.
++      "C4619"
++      "C4623" # default constructor was implicitly defined as deleted
++      "C4625" # copy constructor could not be generated because a base class
++              # copy constructor is inaccessible or deleted
++      "C4626" # assignment operator could not be generated because a base class
++              # assignment operator is inaccessible or deleted
++    "C4643"
++      "C4668" # 'symbol' is not defined as a preprocessor macro, replacing with
++              # '0' for 'directives'
++              # Disable this because GTest uses it everywhere.
++      "C4706" # assignment within conditional expression
++      "C4710" # 'function': function not inlined
++      "C4711" # function 'function' selected for inline expansion
++      "C4800" # 'int' : forcing value to bool 'true' or 'false'
++              # (performance warning)
++      "C4820" # 'bytes' bytes padding added after construct 'member_name'
++    "C4868"
++    "C4996"
++      "C5026" # move constructor was implicitly defined as deleted
++      "C5027" # move assignment operator was implicitly defined as deleted
++      "C5031"
++    "C5039"
++      "C5045"
++      )
++      string(REPLACE "C" " -wd" MSVC_DISABLED_WARNINGS_STR
++                            ${MSVC_DISABLED_WARNINGS_LIST})
++
++      set(CMAKE_CXX_FLAGS " ${CMAKE_CXX_FLAGS} ${MSVC_DISABLED_WARNINGS_STR}")
++ endif()
+diff --git a/include/libcmaes/eigenmvn.h b/include/libcmaes/eigenmvn.h
+index fb8b11d..6a6943b 100644
+--- a/include/libcmaes/eigenmvn.h
++++ b/include/libcmaes/eigenmvn.h
+@@ -58,7 +58,20 @@ public:
+ 	static std::mt19937 rng;                        // The uniform pseudo-random algorithm
+ 	mutable std::normal_distribution<Scalar> norm; // gaussian combinator
+ 	
+-	EIGEN_EMPTY_STRUCT_CTOR(scalar_normal_dist_op)
++	scalar_normal_dist_op() = default;
```

Comment:
I don't see this change referenced anywhere in the logs/upstream. Is this an issue on the library itself? If this is a backport, it'd be great to separate it to its own patch, so that we can crossreference back to it

Reply:
This is removed in the current eigen master branch https://gitlab.com/libeigen/eigen/-/merge_requests/946 

I also got some problems because this additonally defines an empty copy ctr and we needed it to copy the members